ddprof-lib/src/main/cpp/callTraceStorage.cpp

```diff
@@ -84,10 +85,23 @@ HazardPointer::HazardPointer(CallTraceHashTable* resource) : _active(true), _my_
     __atomic_store_n(&global_hazard_list[_my_slot].pointer, resource, __ATOMIC_RELEASE);
 }
 
+// Fast-path constructor using pre-allocated slot (thread-local caching optimization)
+HazardPointer::HazardPointer(CallTraceHashTable* resource, int slot) : _active(true), _my_slot(slot) {
+    // CRITICAL ORDERING: Set bitmap bit BEFORE storing pointer
```

Comment:
I am not sure if I understand correctly, but it sounds to me like you want to do it the other way around: store the pointer first, and only then store the bitmap. Only this will guarantee that other threads see the pointer after they observed the bitmap bit. Or what am I missing?